src/neurostatslib/basis.py

```diff
@@ -1,234 +1,435 @@
-"""Bases classes.
-"""
+"""Bases classes."""
+
 # required to get ArrayLike to render correctly, unnecessary as of python 3.10
 from __future__ import annotations
 
 import abc
-import warnings
-from typing import Tuple
+from typing import Generator, Tuple
 
 import numpy as np
 import scipy.linalg
 from numpy.typing import NDArray
 
+from neurostatslib.utils import row_wise_kron
+
+__all__ = [
+    "MSplineBasis",
+    "RaisedCosineBasisLinear",
+    "RaisedCosineBasisLog",
+    "OrthExponentialBasis",
+    "AdditiveBasis",
+    "MultiplicativeBasis",
```

Comment:
I included `AdditiveBasis` and `MultiplicativeBasis` because otherwise it will interfere with the behavior of inspect (like inspect.get_members). Basically when you list the concrete classes you'll miss add and mult.

Reply:
not sure I understand your comment -- when I type `nsl.basis` and hit tab, I also see `AdditiveBasis` and `MultiplicativeBasis`. You're saying there's no way to avoid that and still have `inspect.get_members` work in the way you rely on in your tests?